app/src/main/java/org/mozilla/fenix/search/awesomebar/AwesomeBarView.kt

```
@@ -0,0 +1,207 @@
+    private val container: ViewGroup,
+    val interactor: AwesomeBarInteractor
+) : LayoutContainer {
+    val view: BrowserAwesomeBar = LayoutInflater.from(container.context)
+        .inflate(R.layout.component_awesomebar, container, true)
+        .findViewById(R.id.awesomeBar)
+
+    override val containerView: View?
+        get() = container
+
+    private val clipboardSuggestionProvider: ClipboardSuggestionProvider
+    private val sessionProvider: SessionSuggestionProvider
+    private val historyStorageProvider: HistoryStorageSuggestionProvider
+    private val shortcutsEnginePickerProvider: ShortcutsSuggestionProvider
+    private val bookmarksStorageSuggestionProvider: BookmarksStorageSuggestionProvider
+    private val defaultSearchSuggestionProvider: SearchSuggestionProvider
+
+    private val loadUrlUseCase = object : SessionUseCases.LoadUrlUseCase {
+        override fun invoke(url: String, flags: EngineSession.LoadUrlFlags) {
+            interactor.onUrlTapped(url)
+        }
+    }
+
+    private val searchUseCase = object : SearchUseCases.SearchUseCase {
+        override fun invoke(searchTerms: String, searchEngine: SearchEngine?) {
+            interactor.onSearchTermsTapped(searchTerms)
+        }
+    }
+
+    private val shortcutSearchUseCase = object : SearchUseCases.SearchUseCase {
+        override fun invoke(searchTerms: String, searchEngine: SearchEngine?) {
+            interactor.onSearchTermsTapped(searchTerms)
+        }
+    }
+
+    init {
+        with(container.context) {
+            val primaryTextColor = ContextCompat.getColor(
+                this,
+                ThemeManager.resolveAttribute(R.attr.primaryText, this)
+            )
+
+            val draw = getDrawable(R.drawable.ic_link)
+            draw?.setColorFilter(primaryTextColor, SRC_IN)
+
+            clipboardSuggestionProvider = ClipboardSuggestionProvider(
+                this,
+                loadUrlUseCase,
+                draw!!.toBitmap(),
```

I'd fail faster if this is null. That way the error falls on the line of code attempting to grab the drawable. `val draw = getDrawable(R.drawable.ic_link)!!`